diff --git a/node_modules/draft-js/lib/DraftEditorLeaf.react.js b/node_modules/draft-js/lib/DraftEditorLeaf.react.js
index abbd02f..89ac6fb 100644
--- a/node_modules/draft-js/lib/DraftEditorLeaf.react.js
+++ b/node_modules/draft-js/lib/DraftEditorLeaf.react.js
@@ -140,6 +140,11 @@ var DraftEditorLeaf = function (_React$Component) {
         mergedStyles.textDecoration = [map.textDecoration, style.textDecoration].join(' ').trim();
       }
 
+      if (style !== undefined && map.className !== style.className) {
+        // .trim() is necessary for IE9/10/11 and Edge
+        mergedStyles.className = [map.className, style.className].join(' ').trim();
+      }
+
       return _assign(map, style, mergedStyles);
     }, {});
 
@@ -148,10 +153,16 @@ var DraftEditorLeaf = function (_React$Component) {
       styleObj = _assign(styleObj, newStyles);
     }
 
+    var className = styleObj.className;
+    if (className) {
+      styleObj.className = undefined;
+    }
+
     return React.createElement(
       'span',
       {
         'data-offset-key': offsetKey,
+        className: className,
         ref: function ref(_ref) {
           return _this2.leaf = _ref;
         },
